src/ucs/arch/x86_64/atomic.h

```diff
@@ -50,4 +50,17 @@
         return prev; \
     }
 
+#define UCS_DEFINE_ATOMIC_BOOL_CSWAP(wordsize, suffix) \
+    static inline uint##wordsize##_t ucs_atomic_bool_cswap##wordsize(volatile uint##wordsize##_t *ptr, \
+                                                                     uint##wordsize##_t compare, \
+                                                                     uint##wordsize##_t swap) { \
+        unsigned long prev; \
+        asm volatile ( \
+              "lock cmpxchg" # suffix " %1, %2" \
+              : "=a" (prev) \
+              : "r"(swap), "m"(*ptr), "0" (compare) \
+              : "memory"); \
+        return prev == compare; \
+    }
+
```

Comment:
may be just define function:
```
static inline uint##wordsize##_t ucs_atomic_bool_cswap##wordsize(volatile uint##wordsize##_t *ptr, \
                                                                     uint##wordsize##_t compare, \
                                                                     uint##wordsize##_t swap) { \
        return ucs_atomic_cswap##wordsize(ptr, compare, swap) == compare; \
    }
```
and may be it will work for all arch's?